superset/distributed_lock/__init__.py

```diff
@@ -75,28 +53,25 @@ def KeyValueDistributedLock(  # pylint: disable=invalid-name
     :yields: A unique identifier (UUID) for the acquired lock (the KV key).
     :raises CreateKeyValueDistributedLockFailedException: If the lock is taken.
     """
+
     # pylint: disable=import-outside-toplevel
-    from superset.commands.key_value.create import CreateKeyValueCommand
-    from superset.commands.key_value.delete import DeleteKeyValueCommand
-    from superset.commands.key_value.delete_expired import DeleteExpiredKeyValueCommand
+    from superset.commands.distributed_lock.create import CreateDistributedLock
+    from superset.commands.distributed_lock.delete import DeleteDistributedLock
+    from superset.commands.distributed_lock.get import GetDistributedLock
 
     key = get_key(namespace, **kwargs)
+    value = GetDistributedLock(namespace=namespace, params=kwargs).run()
+    if value:
+        logger.debug("Lock on namespace %s for key %s already taken", namespace, key)
+        raise CreateKeyValueDistributedLockFailedException("Lock already taken")
+
```

Comment:
I added a get before trying to create the lock, as I noticed that creating locks when it was already taken was producing ugly error logs on Postgres. While this adds one more roundtrip to the metastore, the locking operations are fairly rare, hance it felt cleaner to do it this way. Note, that we're still handling the rare case that the lock got created after the get, so that will still be handled gracefully.